src/as-validator.c

```diff
@@ -1924,6 +1924,49 @@ as_validator_check_branding (AsValidator *validator, xmlNode *node)
 	}
 }
 
+/**
+ * as_validator_check_custom:
+ **/
+static void
+as_validator_check_custom (AsValidator *validator, xmlNode *node)
+{
+	g_autoptr(GHashTable) known_keys = NULL;
+	known_keys = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, NULL);
+
+	for (xmlNode *iter = node->children; iter != NULL; iter = iter->next) {
+		if (iter->type != XML_ELEMENT_NODE)
+			continue;
+
+		g_autofree gchar *node_content = NULL;
+		const gchar *node_name;
+
+		node_name = (const gchar*) iter->name;
+
+		if (g_strcmp0 (node_name, "value") != 0) {
+			as_validator_add_issue (validator, iter, "custom-invalid-tag", node_name);
+			continue;
+		}
+
+		gchar *key_name = NULL;
+		key_name = as_xml_get_prop_value (iter, "key");
+
+		if (key_name == NULL) {
+			as_validator_add_issue (validator, iter, "custom-key-missing", NULL);
+			continue;
+		}
+
+		if (g_hash_table_contains (known_keys, key_name))
+			as_validator_add_issue (validator, iter, "custom-key-duplicated", key_name);
+		else
+			g_hash_table_add (known_keys, key_name);
+
+		node_content = (gchar*) xmlNodeGetContent (iter);
```

Comment:
Please use either `as_xml_get_node_value` or `as_xml_get_node_value_raw`.
You probably want the former and then check for NULL.

Reply:
This unfortunately didn't work
```c
node_content =  as_xml_get_node_value (iter);
if (node_content == NULL) {
    as_validator_add_issue (validator, iter, "custom-value-missing", NULL);
}
```

Reply:
What do you mean it did not work?
This code should be fine:
```c
node_content =  as_xml_get_node_value (iter);
if (node_content == NULL || node_content[0] == '\0')
    as_validator_add_issue (validator, iter, "custom-value-missing", NULL);
```